### ISSUE TITLE
Bug: There's no alphabetical order for workspaces and after creating new workspaces on refresh their order changes reported by @Puneet-here

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11690]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to settings > Create 3 workspaces "A", "B", "C"
2. Go to settings again
3. Refresh

## Expected Result:
They should be in alphabetical order or they should be sorted by time of creation, the order shouldn't change on refresh for the first time.

## Actual Result:
There's no alphabetical order for workspaces and after creating new workspaces on refresh their order changes

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2..12-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/194736941-e195df37-556d-4fcf-b809-a588b920a330.mov

https://user-images.githubusercontent.com/43996225/194736945-da7fbf9a-a952-453b-8dc1-b87cf4bf99a2.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Puneet-here
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665261839338129

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by NicMendonca** [comment]
So on OldDot policies are sorted alphabetically, and I think that makes most sense 👍

**Comment by Puneet-here** [comment]
### Proposal

We need to use 
`.sortBy(policy => policy.title)` before `.value()` at line 203
https://github.com/Expensify/App/blob/dfb5bca05a5b12e62ec86520819fd5a4ac66e405/src/pages/settings/InitialSettingsPage.js#L186-L203

**Comment by MandarDevarshi** [comment]
# Proposal

The solution provided above by Puneet does not work as sorting needs to happen before mapping although no doubt of it being the correct way.

---
## Fix 

By placing `.sortBy(policy => policy.name)`  `after .filter()` solves the issue and it also works if placed `before .filter()` but I am assuming it is better to sort after filter rather than before.

> I tested it and the fix was working, but could not record it because there is a rolling power-cut where I live for a few hours. Would update this comment with recordings then.   
### Before
### After

[My Upwork Profile](https://www.upwork.com/freelancers/~01c1af53a71eb0a42e)

**Comment by bfitzexpensify** [comment]
Upwork job [here](https://www.upwork.com/jobs/~01ce11196cd6582d73)

Reporting job [here](https://www.upwork.com/jobs/~013c75f236a5ea0152) @Puneet-here

**Comment by Puneet-here** [comment]
> The solution provided above by Puneet does not work as sorting needs to happen before mapping although no doubt of it being the correct way.

It's working correctly. @MandarDevarshi, what do you mean by it doesn't work?

**Comment by MandarDevarshi** [comment]
For me on `Web version 1.2.12-3`.

When sortBy is in `line 203` i.e after map. The following is the result.

https://user-images.githubusercontent.com/34445750/195347916-15db7f85-7c59-4e9d-87ee-f0bf0f03b81f.mp4



When sortBy is in `line  188` i.e after filter. The following is the result.

https://user-images.githubusercontent.com/34445750/195348311-be890ef4-d583-4f46-8d49-d880b977aa1c.mp4





What am I missing ? New to expensify, so I could trip as well. :sweat_smile:

**Comment by jasperhuangg** [comment]
@Puneet-here @MandarDevarshi thanks for the discussion. I like where we're going with this.

I'm thinking if we can actually sort the data before it goes into Onyx as opposed to after it's been read from Onyx. This ensures any other functionality we build that involves listing out all of a user's workspaces will be in the same order. 

This might involve making the issue an internal issue. Going to ask for some further clarification here.

**Comment by jasperhuangg** [comment]
@Puneet-here @MandarDevarshi Let's actually go with Puneet's solution since it also handles sorting workspaces after they're created.

**Comment by thesahindia** [comment]
Yeah let's go with @Puneet-here's [proposal](https://github.com/Expensify/App/issues/11690#issuecomment-1274717852). It is working as expected.

C+ reviewed 🎀👀🎀

cc: @jasperhuangg

**Comment by bfitzexpensify** [comment]
Deployed to staging.

**Comment by bfitzexpensify** [comment]
All payments complete. Thanks everyone!
